contracts/token/BiconomyTokenPaymaster.sol

```diff
@@ -395,17 +393,11 @@ contract BiconomyTokenPaymaster is
                 bytes1(paymasterAndData[VALID_PND_OFFSET - 1:VALID_PND_OFFSET])
             )
         );
-        (
-            validUntil,
-            validAfter,
-            feeToken,
-            oracleAggregator,
-            exchangeRate,
-            priceMarkup
-        ) = abi.decode(
-            paymasterAndData[VALID_PND_OFFSET:SIGNATURE_OFFSET],
-            (uint48, uint48, address, address, uint256, uint32)
-        );
+        validUntil = uint48(bytes6(paymasterAndData[21:27]));
+        validAfter = uint48(bytes6(paymasterAndData[27:33]));
+        feeToken = address(bytes20(paymasterAndData[33:53]));
+        exchangeRate = uint128(bytes16(paymasterAndData[53:69]));
+        priceMarkup = uint32(bytes4(paymasterAndData[69:73]));
```

Comment:
Do you want to parameterize the constants?

Reply:
done